src/variational/advi.jl

```diff
@@ -68,7 +68,7 @@ end
 
 Creates a mean-field approximation with multivariate normal as underlying distribution.
 """
-function meanfield(model::Model)
+function meanfield(model::Model; rng::AbstractRNG = GLOBAL_RNG)
```

Comment:
The common approach (as far as I have seen in Random, Distributions and some other packages) is to allow a random number generator as first argument, i.e., to define
```julia
meanfield(model::Model) = meanfield(Random.GLOBAL_RNG, model)
function meanfield(rng::AbstractRNG, model::Model)
    ...
end
```
I'm wondering if that could/should be done here as well?

Reply:
Yeah, so this is what I do for `ELBO` (see `variational/objectives.jl`) and so on. I didn't do it with `meanfield` because I initially didn't think it necessary. What say you @xukai92 ? Should I change from kwarg to this?

Reply:
Non explictly, but we current do both for two different situations
- For internal functions, we stick to the way @devmotion describes above
- For functions to end-users, we use keyword arguments

Reply:
> For functions to end-users, we use keyword arguments

I'm curious, which functions are you referring to? At least the user-facing `sample`, `psample`, and `rand` only allow a random number generator as first argument but not as keyword argument.

Reply:
IIRC our `sample` takes a keyword for `rng`

Reply:
That's not case, I guess that might have changed at some point. The current API of AbstractMCMC and the implementation of `sample` in Turing (https://github.com/TuringLang/Turing.jl/blob/9dc8a297aaed4e898beba698491116c28e40c533/src/inference/Inference.jl#L143-L167) do not take a random number generator as keyword argument but only as first argument.

Reply:
BTW I just notice that this implementation is quite inconsistent since only in the second case the `resume_from` keyword is respected. I'll change that in the updates for AbstractMCMC 0.4.

Reply:
OK cool. If all interfaces take this convention now we should stick to it. Thanks for updating me on this. @torfjelde Let's make it the optional first argument :)

Reply:
Grreaaat! Done!